### PR TITLE
adapter: Add more metrics

### PR DESCRIPTION
**Pull request by ParkMyCar**
In response to incident_74 we want to add more metrics around how long it takes us to query CRDB. This PR adds the following metrics:

1. Histograms for all of the queries the Stash makes.
2. Histogram for tables writes made by the Coordinator.

It also improves the tagging for the "slow coordinator message" metrics to include the subtype of `Command` that the coordinator is running.

### Motivation

Incident 74

### Checklist

- [ ] This PR has adequate test coverage / QA involvement has been duly considered.
- [ ] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [ ] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [ ] If this PR will require changes to cloud orchestration or tests, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [x] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  - N/a

### CODE REVIEW

File: src/adapter/src/metrics.rs [hunk 1]
```diff
@@ -120,6 +121,12 @@ impl Metrics {
                 help: "Number of optimization notices per notice type.",
                 var_labels: ["notice_type"],
             )),
+            append_table_seconds: registry.register(metric!(
+                name: "mz_append_table_seconds",
```
Comment (mjibson) [inline src/adapter/src/metrics.rs:125]:
```suggestion
                name: "mz_append_table_duration_seconds",
```

Prometheus seems to recommend using duration for this stuff: https://prometheus.io/docs/practices/naming/#metric-names

File: src/stash/src/postgres.rs [hunk 2]
```diff
@@ -399,6 +451,12 @@ impl Metrics {
                 help: "Total number of transaction errors.",
                 var_labels: ["cause"],
             )),
+            query_latency_seconds: registry.register(metric!(
+                name: "mz_query_latency",
```
Comment (mjibson) [inline src/stash/src/postgres.rs:455]:
```suggestion
                name: "mz_query_latency_seconds",
```

**Review by mjibson — APPROVED** [review]

File: src/adapter/src/metrics.rs (hunk 1 above)
Comment (ParkMyCar) [reply src/adapter/src/metrics.rs]: Sweet, renamed!

File: src/stash/src/postgres.rs (hunk 2 above)
Comment (ParkMyCar) [reply src/stash/src/postgres.rs:455]: Done!

**Pull request merged**